patches/tModLoader/Terraria/Recipe.TML.cs

```diff
@@ -460,6 +460,12 @@ public Recipe Register()
 		return this;
 	}
 
+	/// <summary>
+	/// Create the recipe but does not yet register it into the game. Call this at the very beginning when creating a new craft.
+	/// </summary>
+	/// <param name="result">What item will be given when the craft has been completed</param>
```

Comment:
I think it might be important to note here that the result variable is actually indicating the ItemID of the "resulting" item of the recipe.